Mathlib/Topology/ContinuousOn.lean

```diff
@@ -1509,6 +1516,12 @@ theorem ContinuousOn.union_isClosed {X Y : Type*} [TopologicalSpace X] [Topologi
   · refine if hx : x ∈ t then hft x hx else continuousWithinAt_of_not_mem_closure ?_
     rwa [ht.closure_eq]
 
+/-- If a function is continuous on two open sets, it is also continuous on their union. -/
+theorem ContinuousOn.union_isOpen {X Y : Type*} [TopologicalSpace X] [TopologicalSpace Y]
+    {s t : Set X} (hs : IsOpen s) (ht : IsOpen t) {f : X → Y} (hfs : ContinuousOn f s)
+    (hft : ContinuousOn f t) : ContinuousOn f (s ∪ t) :=
+  union_continuousAt hs hfs fun _ hx ↦ IsOpen.continuousOn_iff ht |>.mp hft hx
+
```

Comment:
It might also we worth adding the following lemma whilst we're at it!
```suggestion

theorem ContinouousOn_union_isOpen_iff {X Y : Type*} [TopologicalSpace X] [TopologicalSpace Y]
    {s t : Set X} (hs : IsOpen s) (ht : IsOpen t) {f : X → Y} :
    ContinuousOn f (s ∪ t) ↔ ContinuousOn f s ∧ ContinuousOn f t :=
  ⟨fun hf ↦ continuousOn_of_union hf, fun ⟨hfs, hft⟩ ↦ hfs.union_isOpen hs ht hft⟩
```

Reply:
ContinuousOn.union_isOpen should be called `ContinuousOn.union_of_isOpen` - since `isOpen` is one of the hypotheses.

Reply:
And I believe most other files would name the new lemma `continouousOn_union_isOpen`